src/RadioGroup/Radio.tsx

```diff
@@ -0,0 +1,91 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+import "./Radio.less";
+
+const propTypes = {
+  checked: PropTypes.bool,
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string,
+  disabled: PropTypes.bool,
+  id: PropTypes.string.isRequired,
+  onSelect: PropTypes.func,
+  tabIndex: PropTypes.number,
+  value: PropTypes.any,
+};
+
+const cssClass = {
+  CHECKED: "Radio--checked",
+  CONTAINER: "Radio",
+  DISABLED: "Radio--disabled",
+  INNER: "Radio--inner",
+  LABEL: "Radio--label",
+  OUTER: "Radio--outer",
+};
+
+const LABEL_ID_PREFIX = "Radio--label";
+let nextLabelIDSuffix = 0;
+const getNextLabelID = () => `${LABEL_ID_PREFIX}--${nextLabelIDSuffix++}`;
+
+/**
+ * Single radio button input.
+ * Usually used in a RadioGroup for providing a list of mutually-exclusive options to the user.
+ */
+export default class Radio extends React.PureComponent {
+  static propTypes = propTypes;
+  static cssClass = cssClass;
+
+  _element = null;
+  _labelID = getNextLabelID();
+
+  focus() {
+    if (this._element) {
+      this._element.focus();
+    }
+  }
+
+  componentDidMount() {
+    this._element = ReactDOM.findDOMNode(this);
+  }
+
+  render() {
+    const {checked, children, className, disabled, tabIndex} = this.props;
+
+    return (
+      <button
```

Comment:
Why is this a button instead of an input type=radio?

Reply:
it's not possible to arbitrarily style certain native browser elements like the radio button and checkbox, so a common pattern is to use custom elements and assign a11y roles to them so they're still understood by tools like screen readers.

some custom implementations will often create a hidden radio input in addition to this and set its value under the covers to enable native form submit behaviour (i think material does this) but not really necessary in our case, since we're exclusively using AJAX requests to submit form data